### DISCUSSION TITLE
HBRW-009 — Formulae are not Signed

### DISCUSSION THREAD

**Discussion by eszense** [Homebrew/discussions#4110] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
.
```

### Output of `brew doctor`

```shell
.
```

### Description of issue

I read with interest the [security audit report](https://brew.sh/assets/pdf/security_audit_report.pdf) on home brew. I wonder what is the current status regarding the recommendations in `HBRW-009 — Formulae are not Signed`.

In particular, is it (or will it be) possible for the user to cryptographically verify... (e.g. by GPG signature)

1. The formulae to be executed is signed by a trusted homebrew maintainer. 
2. The bottle downloaded (including via mirror) is indeed built using that formulae.
3. The bottle is built by an TestBot ran on a build system controlled by the homebrew team (in contrast to the workstation of a developer/maintainer which is more easily compromised than an ephemeral buildbot.)

Thank you very much.

**Comment by SMillerDev** [comment]
The commits are now signed for Homebrew-core. And work has been done to avoid loading all formulae on every execution of brew commands.
